src/machine/machine_rp2_pll.go

```diff
@@ -98,3 +101,179 @@ func (pll *pll) init(refdiv, vcoFreq, postDiv1, postDiv2 uint32) {
 	pll.pwr.ClearBits(rp.PLL_SYS_PWR_POSTDIVPD)
 
 }
+
+var errVCOOverflow = errors.New("VCO calculation overflow; use lower MHz")
+
+// pllSearch enables searching for a good PLL configuration.
+// Example for 12MHz crystal and RP2040:
+//
+//	fbdiv, refdiv, pd1, pd2, _ := pllSearch{LockRefDiv:1}.CalcDivs(12*MHz, 125*MHz, MHz)
+//
+// Example for 12MHz crystal and RP2350:
+//
+//	fbdiv, refdiv, pd1, pd2, _ := pllSearch{LockRefDiv:1}.CalcDivs(12*MHz, 133*MHz, MHz)
+type pllSearch struct {
+	LowerVCO   bool
+	LockRefDiv uint8
+}
+
+func (ps pllSearch) CalcDivs(xoscRef, targetFreq, MHz uint64) (fbdiv uint64, refdiv, pd1, pd2 uint8, err error) {
```

Comment:
I tried to add a function that dynamically calculates the PLL dividers and optimized enough so that maybe  TinyGo interp would kick in and solve for the dividers at compile time with the XoscFreq and target CPU frequency as solving parameters but it seems I was unsucessful as far as I could tell.

Reply:
I can hardly say I tested it. I compared the results of this function with the ones of vcocalc.py for a few inputs and saw they matched. I left it in in case someone wanted to try to use it in the future seeing how many PRs included xosc changes for the rp2350 recently.

Reply:
Do we have a "scripts" place in tinygo main repo we could put this in and write tests for?

Reply:
> Do we have a "scripts" place in tinygo main repo we could put this in and write tests for?

We have the `tests` directory which is where all of the various testing has gone.

I would suggest we remove that code from this PR, and add it instead to a new draft PR that can be worked on further as part of implementing speed changes.

Reply:
@soypat what do you think?

Reply:
I agree that `CalcDivs` is useful code, in particular for future clock changes. How about moving the code to a `pll_test.go` file and add a test that verifies that the (otherwise magic) pll constants match the output of `ClockDivs`? A future PR would then change, say, `targetFreq`, get a failing test and adjust the other constants according to `ClockDivs`.

Reply:
@eliasnaur that sounds like a very cool idea. Perhaps you could create a PR off of this one to make that change, if you think you see a straightforward way to do it?

Reply:
I don't think it's that complicated, but sure, if @soypat doesn't do it as part of this PR, I can can do it after merge.